### PR TITLE
Add support for Terraform v1.x

### PR DESCRIPTION
**Pull request by soerenmartius**
- feat: upgrade terratest to v0.37.0
- feat: add support for Terraform v1.x
- docs: remove example code from README.md's in examples/
- chore: prepare v0.6.0 release

### CODE REVIEW

**Comment by soerenmartius** [comment]
Terraform seems to fail here @mariux  - could you check on that?

File: examples/basic-routing/README.md [hunk 1]
```diff
@@ -12,47 +12,6 @@ The code in [main.tf] creates a Route53 Zone and Records for the main domain and
...
 The `www.` subdomains are implement through CNAMES and point on the A-Records.
 
-```hcl
-module "route53" {
-  source  = "mineiros-io/route53/aws"
-  version = "~> 0.5.0"
-
-  name = "a-dev-mineiros.io"
-
-  records = [
```
Comment (mariux) [inline examples/basic-routing/README.md:22]: why remove this?

File: test/unit_disabled_test.go [hunk 2]
```diff
@@ -1,27 +0,0 @@
...
-		TerraformDir: "unit-disabled",
-		Vars: map[string]interface{}{
-			"aws_region": "us-east-1",
-		},
-		Upgrade: true,
-	}
-
-	stdout := terraform.InitAndPlan(t, terraformOptions)
-
-	resourceCount := terraform.GetResourceCount(t, stdout)
```
Comment (mariux) [inline test/unit_disabled_test.go:23]: we shoudl not remove the disabled test...

**Review by mariux — CHANGES_REQUESTED** [review]
please re add disabled test

**Comment by mariux** [comment]
cc @soerenmartius

File: examples/basic-routing/README.md (hunk 1 above)
Comment (soerenmartius) [reply examples/basic-routing/README.md:22]: There is 0 value in duplicating code among the `README.md` and the `main.tf` files, it just increases complexity. In newer modules we got rid of that habit also.

**Comment by soerenmartius** [comment]
updated the failing test instead of removing it, r4r

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**